backend/src/Service/Order/OrderService.php

```diff
@@ -325,4 +327,70 @@ public function updateCaptainArrived(OrderUpdateCaptainArrivedRequest $request):
      
         return $this->autoMapping->map(OrderEntity::class, OrderUpdateCaptainArrivedResponse::class, $order);
     }
+
+    public function orderCancel(int $id): ?OrderCancelResponse
+    {
+        $order = $this->orderManager->getOrderById($id);
+     
+        if($order) {
+            $halfHourLaterTime = date_modify($order->getCreatedAt(), '+30 minutes');
+
+            $nowDate = new DateTime('now');
+            
+            if ( $halfHourLaterTime < $nowDate) {
+               //create local notification to store
+               $this->notificationLocalService->createNotificationLocal($order->getStoreOwner()->getStoreOwnerId(), NotificationConstant::CANCEL_ORDER_TITLE, NotificationConstant::CANCEL_ORDER_ERROR_TIME, $order->getId());
```

Comment:
Just a question, why we send notifications when we won't cancel an order (either when time is exceeded, or when captain received the order)?